certbot/certbot/_internal/main.py

```diff
@@ -1559,11 +1567,21 @@ def certonly(config: configuration.NamespaceConfig, plugins: plugins_disco.Plugi
 
     lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
 
+    # If a new cert was issued and we were passed an installer, we can safely
+    # run `installer.restart()` to load the newly issued certificate
+    installer_err: Optional[errors.Error] = None
```

Comment:
How are we feeling about the case where `--certonly` is called but `--nginx/--apache` is set interactively? Because currently this change only catches when `--nginx/--apache` is set on the command line, due to `req_inst` [being set](https://github.com/certbot/certbot/blob/86406ab63aebf463ca4aa0381a55ddeb91231cd2/certbot/certbot/_internal/plugins/selection.py#L210) in `choose_configurator_plugins`. It's probably possible to take a dynamically-selected nginx/apache authenticator and grab the relevant installer based on that, but that's a slightly more extensive change (maybe over in `plugins.selection` instead of here).

Reply:
oh ho ho i didn't know there was an interactive mode. i think it'd be good to make the behavior consistent regardless of how the installer is set

Reply:
Ah yes, while our `certbot.eff.org` instructions and most other advice out there do recommend specifying on the command line, if you run without we'll try to ask interactively what you want it to be. Anyway yeah I agree; without that change we'll hit most use cases, but probably best to be consistent.